api/utils.py

```diff
@@ -0,0 +1,109 @@
+import flask
+import requests
+from flask_smorest import abort
+
+
+def build_request(task_data):
+    """Take data and return an object to PUT to Toolhub"""
+    field = task_data["field"]
+    value = task_data["value"]
+    comment = f"Updated {field} using Toolhunt"
+    data = {}
+    data[field] = value
+    data["comment"] = comment
+    return data
+
+
+def get_current_user():
+    """Get the username of currently logged-in user."""
+    # This import is still throwing an error for me when I put it at the top of the file
+    from app import oauth  # noqa
+
+    if not flask.session:
+        abort(401, message="No user is currently logged in.")
+    else:
+        try:
+            resp = oauth.toolhub.get("user/", token=flask.session["token"])
+            print(resp, "This is from the function")
+            resp.raise_for_status()
+            profile = resp.json()
+            username = profile["username"]
+            return username
+        except requests.exceptions.HTTPError as err:
+            print(err)
+            abort(401, message="User authorization failed.")
+        except requests.exceptions.ConnectionError as err:
+            print(err)
+            abort(503, message="Server connection failed.  Please try again.")
+        except requests.exceptions.RequestException as err:
+            print(err)
+            abort(501, message="Server encountered an unexpected error.")
+
+
+class ToolhubClient:
```

Comment:
Let's decouple this Class from the app context and let the user pass in Toolhub URL and headers.